Source/Dafny/Rewriter.cs

```diff
@@ -1498,9 +1498,15 @@ public TimeLimitRewriter(ErrorReporter reporter)
 
     internal override void PreResolve(ModuleDefinition m) {
       foreach (var d in m.TopLevelDecls) {
-        if (d is ClassDecl) {
-          var c = (ClassDecl)d;
```

Comment:
The proposed fix still misses `codatatype` and `newtype` declarations. Go back to the original code and just change `ClassDecl` on L1501 and L1502 to `TopLevelDeclWithMembers`.

Reply:
Thanks for the pointer to `TopLevelDeclWithMembers`.  That does look like a cleaner solution.